source/common/http/conn_manager_impl.cc

```diff
@@ -716,15 +716,16 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(ActiveStreamDecoderFilte
     entry = std::next(filter->entry());
   }
 
-  for (; entry != decoder_filters_.end(); entry++) {
+  for (; entry != decoder_filters_.end() && !decoding_headers_only_; entry++) {
```

Comment:
Is this correct? I think you still need to call remaining filters with decode/encodeHeaders with end_stream true?

Reply:
Not clear to me which one is the more correct. As written this change stops filter iteration completely, but I can see why you'd want subsequent filters to allow modifying the headers as well. Happy to change it if the latter makes more sense in the context of the rest of the filter API.

If we're gonna continue the filters I think we'd need to rename the enum return value, maybe something like `ContinueHeadersOnly`?

Reply:
+1